### PR TITLE
Fixes #21849 - Check if the user is new before disabling login

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

**Review by tbrisker — CHANGES_REQUESTED** [review]
Can you please add an integration test for creating a new user so that we don't break it again?

**Comment by shiramax** [comment]
@tbrisker Done.

File: app/views/users/_form.html.erb [hunk 1]
```diff
@@ -20,7 +20,9 @@
 
   <div class='tab-content'>
     <div class='tab-pane active' id='primary'>
-      <%  msg =  @user.internal? ? "": "External user login can not be changed" %>
+      <% if !(@user.internal?) && !(@user.new_record? ) %>
```
Comment (iNecas) [inline app/views/users/_form.html.erb]: The parenthesis are not needed here (or at least the spacing should be consistent

**Review by iNecas — CHANGES_REQUESTED** [review]

**Review by tbrisker — APPROVED** [review]

**Comment by tbrisker** [comment]
[test foreman]

**Review by iNecas — APPROVED** [review]

**Comment by tbrisker** [comment]
Test failures are intermittent connection issues: 
```
Capybara::Poltergeist::StatusFailError: Request to 'http://127.0.0.1:37596/hosts/host26/edit' failed to reach server, check DNS and/or server status
    test/integration/host_js_test.rb:586:in `block (3 levels) in <class:HostJSTest>'
    test/test_helper.rb:35:in `before_setup' (Capybara::Poltergeist::StatusFailError)
```
merging despite test failure.
Thanks @shiramax !

**Pull request merged**